app/models/miq_worker/runner.rb

```diff
@@ -374,6 +375,32 @@ def heartbeat_to_drb
   def heartbeat_to_file(timeout = nil)
     timeout ||= worker_settings[:heartbeat_timeout] || Workers::MiqDefaults.heartbeat_timeout
     File.write(@worker.heartbeat_file, (Time.now.utc + timeout).to_s)
+
+    get_messages.each { |msg, *args| process_message(msg, *args) }
+  end
+
+  def get_messages
+    messages = []
+    now = Time.now.utc
+    @my_last_config_change ||= now
+
+    last_config_change = server_last_change(:last_config_change)
+    if last_config_change && last_config_change > @my_last_config_change
+      _log.info("#{log_prefix} Configuration has changed, New TS: #{last_config_change}, Old TS: #{@my_last_config_change}")
+      messages << ["sync_config"]
+
+      @my_last_config_change = last_config_change
+    end
+
+    messages
+  end
+
+  def key_store
+    @key_store ||= Dalli::Client.new(MiqMemcached.server_address, :namespace => "server_monitor")
+  end
```

Comment:
Most of that work should be handled here:  https://github.com/ManageIQ/manageiq/pull/15818

And that said, I don't know that there are any messages we would have to worry about doing this for after the above, so I would agree that my suggestion is probably premature knowing that.